membership-attribute-service/app/controllers/SalesforceHookController.scala

```diff
@@ -40,99 +40,153 @@ class SalesforceHookController extends LazyLogging {
     </soapenv:Envelope>
   )
 
-def createAttributes = BackendFromSalesforceAction.async(parse.xml) { request =>
-
-  val touchpoint = request.touchpoint
-  val validOrgId = touchpoint.sfOrganisationId
-  val attributeService = touchpoint.attrService
-  val requestId = scala.util.Random.nextInt
-
-  implicit val pf = Membership
-
-  logger.info(s"Called from Saleforce to modify the Members Data API. Request Id: $requestId")
-
-  def deleteMemberRecord(membershipDeletion: MembershipDeletion): Future[Object] = {
-    val userId = membershipDeletion.userId
-    attributeService.delete(userId).map { deleteItemResult =>
-      logger.info(s"Successfully deleted user $userId from ${touchpoint.dynamoAttributesTable}.")
-      metrics.put("Delete", 1)
-      deleteItemResult
-    }.recover { case e: Throwable =>
-      logger.warn(s"Failed to delete user $userId from ${touchpoint.dynamoAttributesTable}. Salesforce should retry.", e)
-      Failure
+  def createAttributes = BackendFromSalesforceAction.async(parse.xml) { request =>
+
+    val touchpoint = request.touchpoint
+    val validOrgId = touchpoint.sfOrganisationId
+    val attributeService = touchpoint.attrService
+
+    object ContextLogging {
+      case class RequestId(value: Int)
+      implicit val requestId = RequestId(scala.util.Random.nextInt)
+      def info(message: String)(implicit requestId: RequestId) =
+        logger.info(s"$requestId: $message")
+      def warn(message: String)(implicit requestId: RequestId) =
+        logger.warn(s"$requestId: $message")
+      def warn(message: String, e: Throwable)(implicit requestId: RequestId) =
+        logger.warn(s"$requestId: $message", e)
+      def error(message: String)(implicit requestId: RequestId) =
+        logger.error(s"$requestId: $message")
+
+      implicit class FutureContextLoggable[T](future: Future[T]) {
+        def logWith(message: String, extractor: T => Any = identity): Future[T] = {
+          future.foreach(any => info(s"$message {${extractor(any)}}"))
+          future
+        }
+
+      }
+      implicit class ContextLoggable[T](t: T) {
+        def logWith(message: String): T = {
+          t match {
+            case future: Future[_] => future.foreach(any => info(s"$message {$any}"))
+            case any => info(s"$message {$any}")
+          }
+          t
+        }
+
+      }
+
     }
-  }
+    import ContextLogging._
+
+    implicit val pf = Membership
 
-  def updateMemberRecord(membershipUpdate: MembershipUpdate): Future[Object] = {
+    info(s"Called from Saleforce to modify the Members Data API")
```

Comment:
we only are logging the dynamo record or information that goes into it.  the extra logging is basically this:
```
INFO: RequestId(-1247831390): Called from Saleforce to modify the Members Data API
INFO: RequestId(-1247831390): Parsed Salesforce message successfully. Salesforce sent 1 objects to update: List(MembershipUpdate(30000594,Supporter,Some(003g000001VLiO6AAL)))
INFO: RequestId(-1247831390): Salesforce called has been parsed. Attrs: MembershipUpdate(30000594,Supporter,Some(003g000001VLiO6AAL))
INFO: RequestId(-1247831390): contact id from SF {Some(003g000001VLiO6AAL)}
INFO: RequestId(-1247831390): current subscriptions {List(Id(2c92c0f95cac811c015cbff6a181648b))}
id='2c92c0f95cac811c015cbff6a14c6487'
INFO: RequestId(-1247831390): ready to update dynamo with Attributes(30000594,Supporter,Some(003g000001VLiO6AAL),None,Some(2),Some(2022),Some(2017-06-19))
INFO: RequestId(-1247831390): Successfully inserted Attributes(30000594,Supporter,Some(003g000001VLiO6AAL),None,Some(2),Some(2022),Some(2017-06-19)) into MembershipAttributes-DEV.

```